src/Roslyn.Diagnostics.Analyzers/UnitTests/AvoidOptSuffixForNullableEnableCodeTests.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Roslyn.Diagnostics.CSharp.Analyzers.CSharpAvoidOptSuffixForNullableEnableCode,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Roslyn.Diagnostics.Analyzers.UnitTests
+{
+    public class AvoidOptSuffixForNullableEnableCodeTests
+    {
+        [Fact]
+        public async Task RS0046_CSharp8_NullableEnabledCode_Diagnostic()
+        {
+            await new VerifyCS.Test
+            {
+                LanguageVersion = LanguageVersion.CSharp8,
+                TestCode =
+                        @"
+#nullable enable
+
+public class Class1
+{
+    private Class1? [|_instanceOpt|];
```

Comment:
We should write a code fix that invokes [Renamer.RenameSymbolAsync](http://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Rename/Renamer.cs,17). I think this is **absolutely** required as most users would prefer to do a FixAll operation to get rid of the opt suffix.